# Importing the libraries

In [1]:
import pandas as pd

# Setting the path and uploading the excel into dataframes

In [2]:
path = "D:\Work\My work\Edulyt Internship\Dataset\Edulyt India - Data for Project - 3.xlsx"

In [3]:
dataset_account = pd.read_excel(path, sheet_name = 'New_Account')
dataset_region = pd.read_excel(path, sheet_name = 'District')
dataset_card = pd.read_excel(path, sheet_name = 'New_Card')
dataset_disposition = pd.read_excel(path, sheet_name = 'New_Disposition')

# Viewing the datasets and Renaming the required columns

In [4]:
dataset_account.head()

account_id  district_id         frequency
0         576           55  MONTHLY ISSUANCE
1        3818           74  MONTHLY ISSUANCE
2         704           55  MONTHLY ISSUANCE
3        2378           16  MONTHLY ISSUANCE
4        2632           24  MONTHLY ISSUANCE

In [5]:
dataset_region.head()

A1           A2               A3       A4  A5  A6  A7  A8  A9    A10  \
0   1  Hl.m. Praha           Prague  1204953   0   0   0   1   1  100.0   
1   2      Benesov  central Bohemia    88884  80  26   6   2   5   46.7   
2   3       Beroun  central Bohemia    75232  55  26   4   1   5   41.7   
3   4       Kladno  central Bohemia   149893  63  29   6   2   6   67.4   
4   5        Kolin  central Bohemia    95616  65  30   4   1   6   51.4   

     A11   A12   A13  A14    A15    A16  
0  12541  0.29  0.43  167  85677  99107  
1   8507  1.67  1.85  132   2159   2674  
2   8980  1.95  2.21  111   2824   2813  
3   9753  4.64  5.05  109   5244   5892  
4   9307  3.85  4.43  118   2616   3040

In [6]:
dataset_card.head()

card_id  disp_id     type  issued
0     1005     9285  CLASSIC  931107
1      104      588  CLASSIC  940119
2      747     4915  CLASSIC  940205
3       70      439  CLASSIC  940208
4      577     3687  CLASSIC  940215

In [7]:
dataset_disposition.head()

disp_id  client_id  account_id   type
0        1          1           1  OWNER
1        2          2           2  OWNER
2        3          3           2   USER
3        4          4           3  OWNER
4        5          5           3   USER

In [8]:
dataset_account.describe()

account_id  district_id
count   4500.000000  4500.000000
mean    2786.067556    37.310444
std     2313.811984    25.177217
min        1.000000     1.000000
25%     1182.750000    13.000000
50%     2368.000000    38.000000
75%     3552.250000    60.000000
max    11382.000000    77.000000

In [9]:
dataset_region.describe()

A1            A4          A5         A6         A7         A8  \
count  77.000000  7.700000e+01   77.000000  77.000000  77.000000  77.000000   
mean   39.000000  1.338849e+05   48.623377  24.324675   6.272727   1.727273   
std    22.371857  1.369135e+05   32.741829  12.780991   4.015222   1.008338   
min     1.000000  4.282100e+04    0.000000   0.000000   0.000000   0.000000   
25%    20.000000  8.585200e+04   22.000000  16.000000   4.000000   1.000000   
50%    39.000000  1.088710e+05   49.000000  25.000000   6.000000   2.000000   
75%    58.000000  1.390120e+05   71.000000  32.000000   8.000000   2.000000   
max    77.000000  1.204953e+06  151.000000  70.000000  20.000000   5.000000   

              A9         A10           A11        A13         A14  \
count  77.000000   77.000000     77.000000  77.000000   77.000000   
mean    6.259740   63.035065   9031.675325   3.787013  116.129870   
std     2.435497   16.221727    790.202347   1.908480   16.608773   
min     1.000000   33.900000   8110.000000   0.430000   81.000000   
25%     5.000000   51.900000   8512.000000   2.310000  105.000000   
50%     6.000000   59.800000   8814.000000   3.600000  113.000000   
75%     8.000000   73.500000   9317.000000   4.790000  126.000000   
max    11.000000  100.000000  12541.000000   9.400000  167.000000   

                A16  
count     77.000000  
mean    5030.831169  
std    11270.796786  
min      888.000000  
25%     2122.000000  
50%     3040.000000  
75%     4595.000000  
max    99107.000000

In [10]:
dataset_card.describe()

card_id       disp_id         issued
count   892.000000    892.000000     892.000000
mean    480.855381   3511.862108  972309.676009
std     306.933982   2984.373626   10606.196816
min       1.000000      9.000000  931107.000000
25%     229.750000   1387.000000  970125.500000
50%     456.500000   2938.500000  980106.000000
75%     684.250000   4459.500000  980805.250000
max    1247.000000  13660.000000  981229.000000

In [11]:
dataset_disposition.describe()

disp_id     client_id    account_id
count   5369.000000   5369.000000   5369.000000
mean    3337.097970   3359.011920   2767.496927
std     2770.418826   2832.911984   2307.843630
min        1.000000      1.000000      1.000000
25%     1418.000000   1418.000000   1178.000000
50%     2839.000000   2839.000000   2349.000000
75%     4257.000000   4257.000000   3526.000000
max    13690.000000  13998.000000  11382.000000

In [12]:
dataset_region.rename(columns = {'A1':'district_id', 'A2':'district_name','A3':'region', 'A11': 'Avg_Salary'}, inplace = True)

In [13]:
dataset_region.describe()

district_id            A4          A5         A6         A7         A8  \
count    77.000000  7.700000e+01   77.000000  77.000000  77.000000  77.000000   
mean     39.000000  1.338849e+05   48.623377  24.324675   6.272727   1.727273   
std      22.371857  1.369135e+05   32.741829  12.780991   4.015222   1.008338   
min       1.000000  4.282100e+04    0.000000   0.000000   0.000000   0.000000   
25%      20.000000  8.585200e+04   22.000000  16.000000   4.000000   1.000000   
50%      39.000000  1.088710e+05   49.000000  25.000000   6.000000   2.000000   
75%      58.000000  1.390120e+05   71.000000  32.000000   8.000000   2.000000   
max      77.000000  1.204953e+06  151.000000  70.000000  20.000000   5.000000   

              A9         A10    Avg_Salary        A13         A14  \
count  77.000000   77.000000     77.000000  77.000000   77.000000   
mean    6.259740   63.035065   9031.675325   3.787013  116.129870   
std     2.435497   16.221727    790.202347   1.908480   16.608773   
min     1.000000   33.900000   8110.000000   0.430000   81.000000   
25%     5.000000   51.900000   8512.000000   2.310000  105.000000   
50%     6.000000   59.800000   8814.000000   3.600000  113.000000   
75%     8.000000   73.500000   9317.000000   4.790000  126.000000   
max    11.000000  100.000000  12541.000000   9.400000  167.000000   

                A16  
count     77.000000  
mean    5030.831169  
std    11270.796786  
min      888.000000  
25%     2122.000000  
50%     3040.000000  
75%     4595.000000  
max    99107.000000

# Merging the dataframes and dropping redundant columns in order to obtain the final dataframe

In [14]:
dataset_region.drop(['A4', 'A5', 'A6', 'A7', 'A8', 'A9', 'A10', 'A12', 'A5', 'A13', 'A14', 'A15', 'A16'], axis = 1, inplace = True)

In [15]:
df1 = dataset_account.merge(dataset_region, how = 'left', on='district_id')

In [16]:
df1

account_id  district_id         frequency      district_name  \
0            576           55  MONTHLY ISSUANCE      Brno - venkov   
1           3818           74  MONTHLY ISSUANCE    Ostrava - mesto   
2            704           55  MONTHLY ISSUANCE      Brno - venkov   
3           2378           16  MONTHLY ISSUANCE  Jindrichuv Hradec   
4           2632           24  MONTHLY ISSUANCE       Karlovy Vary   
...          ...          ...               ...                ...   
4495         124           55  MONTHLY ISSUANCE      Brno - venkov   
4496        3958           59  MONTHLY ISSUANCE           Kromeriz   
4497         777           30  MONTHLY ISSUANCE            Sokolov   
4498        1573           63  MONTHLY ISSUANCE             Vyskov   
4499        3276            1  MONTHLY ISSUANCE        Hl.m. Praha   

             region  Avg_Salary  
0     south Moravia        8743  
1     north Moravia       10673  
2     south Moravia        8743  
3     south Bohemia        8427  
4      west Bohemia        8991  
...             ...         ...  
4495  south Moravia        8743  
4496  south Moravia        8444  
4497   west Bohemia        9650  
4498  south Moravia        8288  
4499         Prague       12541  

[4500 rows x 6 columns]

In [17]:
df2 = dataset_card.merge(dataset_disposition, how = 'left', on = 'disp_id')

In [18]:
df2

card_id  disp_id   type_x  issued  client_id  account_id type_y
0       1005     9285  CLASSIC  931107       9593        7753  OWNER
1        104      588  CLASSIC  940119        588         489  OWNER
2        747     4915  CLASSIC  940205       4915        4078  OWNER
3         70      439  CLASSIC  940208        439         361  OWNER
4        577     3687  CLASSIC  940215       3687        3050  OWNER
..       ...      ...      ...     ...        ...         ...    ...
887      125      694     GOLD  981226        694         577  OWNER
888      674     4360  CLASSIC  981228       4360        3614  OWNER
889      322     2063  CLASSIC  981228       2063        1698  OWNER
890      685     4467  CLASSIC  981229       4467        3705  OWNER
891      635     4083  CLASSIC  981229       4083        3377  OWNER

[892 rows x 7 columns]

In [19]:
df3 = df2.merge(df1, how = 'left', on = 'account_id')

In [20]:
df3

card_id  disp_id   type_x  issued  client_id  account_id type_y  \
0       1005     9285  CLASSIC  931107       9593        7753  OWNER   
1        104      588  CLASSIC  940119        588         489  OWNER   
2        747     4915  CLASSIC  940205       4915        4078  OWNER   
3         70      439  CLASSIC  940208        439         361  OWNER   
4        577     3687  CLASSIC  940215       3687        3050  OWNER   
..       ...      ...      ...     ...        ...         ...    ...   
887      125      694     GOLD  981226        694         577  OWNER   
888      674     4360  CLASSIC  981228       4360        3614  OWNER   
889      322     2063  CLASSIC  981228       2063        1698  OWNER   
890      685     4467  CLASSIC  981229       4467        3705  OWNER   
891      635     4083  CLASSIC  981229       4083        3377  OWNER   

     district_id         frequency      district_name         region  \
0             74  MONTHLY ISSUANCE    Ostrava - mesto  north Moravia   
1             16  MONTHLY ISSUANCE  Jindrichuv Hradec  south Bohemia   
2             40  MONTHLY ISSUANCE            Teplice  north Bohemia   
3             51  MONTHLY ISSUANCE            Trutnov   east Bohemia   
4             60  MONTHLY ISSUANCE          Prostejov  south Moravia   
..           ...               ...                ...            ...   
887           29  MONTHLY ISSUANCE           Rokycany   west Bohemia   
888           19  MONTHLY ISSUANCE         Prachatice  south Bohemia   
889           68  MONTHLY ISSUANCE    Frydek - Mistek  north Moravia   
890           20  MONTHLY ISSUANCE         Strakonice  south Bohemia   
891           74  MONTHLY ISSUANCE    Ostrava - mesto  north Moravia   

     Avg_Salary  
0         10673  
1          8427  
2          9317  
3          8541  
4          8441  
..          ...  
887        8843  
888        8402  
889        9893  
890        8547  
891       10673  

[892 rows x 12 columns]

In [21]:
df3.drop(['disp_id', 'type_x', 'type_y', 'frequency'], axis = 1, inplace=True) #This is the final dataframe we will be working with

In [22]:
df3

card_id  issued  client_id  account_id  district_id      district_name  \
0       1005  931107       9593        7753           74    Ostrava - mesto   
1        104  940119        588         489           16  Jindrichuv Hradec   
2        747  940205       4915        4078           40            Teplice   
3         70  940208        439         361           51            Trutnov   
4        577  940215       3687        3050           60          Prostejov   
..       ...     ...        ...         ...          ...                ...   
887      125  981226        694         577           29           Rokycany   
888      674  981228       4360        3614           19         Prachatice   
889      322  981228       2063        1698           68    Frydek - Mistek   
890      685  981229       4467        3705           20         Strakonice   
891      635  981229       4083        3377           74    Ostrava - mesto   

            region  Avg_Salary  
0    north Moravia       10673  
1    south Bohemia        8427  
2    north Bohemia        9317  
3     east Bohemia        8541  
4    south Moravia        8441  
..             ...         ...  
887   west Bohemia        8843  
888  south Bohemia        8402  
889  north Moravia        9893  
890  south Bohemia        8547  
891  north Moravia       10673  

[892 rows x 8 columns]

# Finding out the districts with average salary > 9000

In [23]:
df4 = df3.copy()
df5 = df4[df4['Avg_Salary']>=9000]

In [24]:
df5

card_id  issued  client_id  account_id  district_id    district_name  \
0       1005  931107       9593        7753           74  Ostrava - mesto   
2        747  940205       4915        4078           40          Teplice   
5        377  940303       2429        1999           70          Karvina   
10       732  940721       4763        3948            1      Hl.m. Praha   
11       181  940819       1066         885            1      Hl.m. Praha   
..       ...     ...        ...         ...          ...              ...   
882      924  981224       7862        6510           74  Ostrava - mesto   
885      753  981225       4974        4131            5            Kolin   
886      442  981226       2796        2310           15    Cesky Krumlov   
889      322  981228       2063        1698           68  Frydek - Mistek   
891      635  981229       4083        3377           74  Ostrava - mesto   

              region  Avg_Salary  
0      north Moravia       10673  
2      north Bohemia        9317  
5      north Moravia       10177  
10            Prague       12541  
11            Prague       12541  
..               ...         ...  
882    north Moravia       10673  
885  central Bohemia        9307  
886    south Bohemia        9045  
889    north Moravia        9893  
891    north Moravia       10673  

[437 rows x 8 columns]

# Printing the results

In [25]:
print("Total number of cards issued = ", len(df3))
print("No. of cards issued for regions where average salary is greater than or equal to 9000 = ", len(df5))
print("% of cards from regions where average salary is greater than or equal to 9000 = ",  len(df5)/len(df3) * 100)

Total number of cards issued =  892
No. of cards issued for regions where average salary is greater than or equal to 9000 =  437
% of cards from regions where average salary is greater than or equal to 9000 =  48.99103139013453


### Conclusion - Thus, only 48.99% of the cards have been issued from regions where the average salary is greater than or equal to 9000, which is less than 50% of all the cards issued. Thus, it is not a good strategy to issue cards only in those regions.